### ISSUE TITLE
Syntax-Highlighting issue

### ISSUE THREAD

**Issue by Hyruleh3ro** [rust-lang/rust-analyzer#14821]
**Code:**

fn main() {
    let heart_eyed_cat = '😻';
    let value_of_heart_eye: i32 = heart_eyed_cat
        .to_string()
        .trim()
        .trim_end()
        .parse()
        .expect("Hi");
    let value_of_heart_eye: i32 = match heart_eyed_cat.to_string().trim().parse() {
        Ok(num) => num,
        Err(_) => todo!(),
    };
}

**Cargo.toml:**

[package]
name = "BasicRust2"
version = "0.1.0"
edition = "2021"

\# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
//(no dependencies added)

**Version of Rust-analyser:**

![Screenshot 2023-05-16 182803](https://github.com/rust-lang/rust-analyzer/assets/73059478/070573da-39a8-4a08-9867-9c624a97d864)

**Bug description:**
Without having executed the code before (fresh VS Code start) rust-analyser doesn't highlight the "Hi: ParseIntError { kind: InvalidDigit }panic"-Error (Picture 1). After I execute the code I receive a compiler error (Picture 2). Afterwards rust-analyser starts highlighting the error, but from then on it doesn't stop highlighting it (Pictures 3-5). Despite me turning the lines in question into a comment and even deleting them, rust-analyser still marks an error even in a empty line. (Picture numbers are each below the image)

![Screenshot (696)](https://github.com/rust-lang/rust-analyzer/assets/73059478/617f3ebb-1ddb-4e17-9b88-601a0a101528)
_Picture 1_

![Screenshot (697)](https://github.com/rust-lang/rust-analyzer/assets/73059478/99447722-ebd1-4e41-adb1-cf296b8cabf1)
_Picture 2_

![Screenshot (698)](https://github.com/rust-lang/rust-analyzer/assets/73059478/32c73e09-d588-4640-a8fc-28c6faf6583a)
_Picture 3_

![Screenshot (699)](https://github.com/rust-lang/rust-analyzer/assets/73059478/898097c9-3f56-46dc-8600-f3e7ca9f068d)
_Picture 4_

![Screenshot (700)](https://github.com/rust-lang/rust-analyzer/assets/73059478/32f8dc89-e843-4215-ac34-4ee9b74d5e73)
_Picture 5_

**Comment by Veykril** [comment]
Does it disappearafter rerunning the task?

**Comment by Hyruleh3ro** [comment]
Yes, I just tested it and indeed the false highlighting disappears after rerunning the program (alternatively after restarting VsCode as well).

**Comment by ChayimFriedman2** [comment]
Not reproducible anymore.
